src/sentry/integrations/slack/utils/users.py

```diff
@@ -18,31 +22,58 @@
 
 
 def get_users(integration: Integration, organization: Organization) -> Sequence[Mapping[str, Any]]:
-    client = SlackClient(integration_id=integration.id)
-
     user_list = []
     next_cursor = None
-    for i in range(SLACK_GET_USERS_PAGE_LIMIT):
+
+    if features.has("organizations:slack-sdk-get-users", organization):
+        sdk_client = SlackSdkClient(integration_id=integration.id)
+
         try:
-            next_users = client.get(
-                "/users.list",
-                params={"limit": SLACK_GET_USERS_PAGE_SIZE, "cursor": next_cursor},
-            )
-        except ApiError as e:
+            users = sdk_client.users_list(cursor=next_cursor, limit=SLACK_GET_USERS_PAGE_SIZE)
+        except SlackApiError as e:
             logger.info(
-                "post_install.fail.slack_users.list",
+                "slack.post_install.get_users.error",
                 extra={
                     "error": str(e),
                     "organization": organization.slug,
                     "integration_id": integration.id,
                 },
             )
-            break
-        user_list += next_users["members"]
+            return []
 
-        next_cursor = next_users["response_metadata"]["next_cursor"]
-        if not next_cursor:
-            break
+        next_users = iter(users)
+        for _ in range(SLACK_GET_USERS_PAGE_LIMIT):
```

Comment:
once we reach the limit, we just end silently? Is that the best corse of action? I think this flow needs to be updated, why not make this return a generator and yield each batch?

Reply:
I know you're just trying to keep the functionality the same as before, but this is a limitation we should address

Reply:
should we have the generator and iterate through until an error happens or we finish?

Reply:
i added a generator function for this